packages/block-editor/src/components/inserter/block-patterns-tab.js

```diff
@@ -49,6 +51,18 @@ function usePatternsCategories( rootClientId ) {
 		rootClientId
 	);
 
+	const [ unsyncedPatterns ] = useBlockTypesState(
+		rootClientId,
+		undefined,
+		'unsynced'
+	);
+
+	const filteredUnsyncedPatterns = useMemo( () => {
+		return unsyncedPatterns.filter(
+			( { category: unsyncedPatternCategory } ) =>
+				unsyncedPatternCategory === 'reusable'
+		);
+	}, [ unsyncedPatterns ] );
```

Comment:
This filtering could probably do with some optimization, as `unsyncedPatterns` seems to contain the inserter data for every block type, which then gets filtered down. There's probably a more direct way to do this that doesn't involved getting other non-pattern block types first.

Seems like a good task for a follow-up though.